Fix quota sync stage

derekwaynecarr opened the pull request:
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1449568

derekwaynecarr commented:
@eparis @deads2k

smarterclayton commented:
LGTM

deads2k commented on vendor/k8s.io/kubernetes/pkg/controller/resourcequota/resource_quota_controller.go:
```diff
@@ -254,6 +254,8 @@ func (rq *ResourceQuotaController) Run(workers int, stopCh <-chan struct{}) {
 		go wait.Until(rq.worker(rq.missingUsageQueue), time.Second, stopCh)
 	}
+	// the timer for how often we do a full recalculation across all quotas
+	go wait.Until(func() { rq.enqueueAll() }, rq.resyncPeriod(), stopCh)
```
Are we intentionally doing this every five minutes?  Isn't that heavy?

deads2k commented on pkg/cmd/server/start/start_master.go:
```diff
@@ -755,7 +755,7 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 
 	// initializes quota docs used by admission
-	oc.RunResourceQuotaManager(nil)
+	oc.RunResourceQuotaManager(controllerManagerOptions)
```
Cleanup this method to prevent nils and respect the config.  I don't see a reason to ever set it without these options.

jupierce commented:
@derekwaynecarr I saw this merge in master. Ok to merge in stage?

derekwaynecarr commented:
@deads2k -- the master pr merged before addressing your comments.  i would like to merge this in stage, and then respond to your comments in master in a follow-on.

[merge]

eparis commented:
[merge][severity: blocker]

(just to get in the mindset to use the severity tags)

deads2k commented:
> [merge][severity: blocker]
> 
> (just to get in the mindset to use the severity tags)

Any chance of using labels instead to align us with upstream?  Also, it's a whole lot easier to see, find, and sort.

eparis commented:
We hope to eventually move to the upstream toolset. For now, this is intended to mimic the (black magic) used by extended tests. Labels were suggested but the tooling could already support this. We are trying to do the minimal amount to survive on the old tooling so can hopefully get to new tooling more quickly.

derekwaynecarr commented:
[merge][severity: blocker]
